feat: add storybook quickstart (#117)
argos-ci/docs: docs/quickstart/storybook.mdx

---
slug: /quickstart/storybook
title: Storybook
sidebar_position: 20
---

import {
  AddSecret,
  Congratulation,
  HelpSection,
  InstallDevDep,
  AddToGitIgnore,
} from "@site/src/partials";

# Storybook Quickstart

Learn how to setup visual testing in a Storybook using Argos.

> The best way to integrate Argos with Storybook is to use [Storycap](https://github.com/reg-viz/storycap) to crawl your storybook and capture screenshots of your components.

Reviewer: ```suggestion
The best way to integrate Argos with Storybook is to use [Storycap](https://github.com/reg-viz/storycap) to crawl your Storybook and capture screenshots of your components.
```

## Prerequisites

> To get the most out of this guide, you’ll need to:

Reviewer: ```suggestion
To get the most out of this guide, you'll need to:
```

> - Have a project with [Storybook](https://storybook.js.org/docs/get-started/install)
> - Run your tests on CI
> - [Create your project in Argos](https://app.argos-ci.com/new)

Reviewer: ```suggestion
- [Use Storybook](https://storybook.js.org/docs/get-started/install)
- [Create your project in Argos](https://app.argos-ci.com/new)
```

Alignment with other docs and no need to "Run your tests on CI" for this one.

## 1. Install

<InstallDevDep dependency="@argos-ci/cli storycap" showCliLink />

## 2. Capture screenshots

There are two ways to capture screenshots of your Storybook:

- If your Storybook is running and accessible via an URL, add this command to your CI pipeline to capture screenshots of your stories:

```sh
# Capture screenshots of your stories

> npm exec -- storycap http://localhost:6006 --outDir ./screenshots

Reviewer: ```suggestion
npm exec -- storycap <STORYBOOK-URL> --outDir ./screenshots
```

```

- If your Storybook is not deployed, you need to serve your Storybook before capturing the screenshots. Use the following commands:

```sh
# Build Storybook
npm exec -- storybook build --output-dir ./storybook-static

# Screenshot Storybook with Storycap

> npm exec -- storycap --serverCmd "npm exec -- http-server ./storybook-static --port 6006" http://127.0.0.1:6006/ --outDir ./screenshots

Reviewer: ```suggestion
npm exec -- storycap --serverCmd "npx http-server ./storybook-static --port 6006" http://127.0.0.1:6006/ --outDir ./screenshots
```
